llama_index/response_synthesizers/base.py

```diff
@@ -44,6 +46,17 @@ def __init__(
         self._streaming = streaming
         self._output_cls = output_cls
 
+    def _get_prompts(self) -> Dict[str, Any]:
+        """Get prompts."""
+        return {}
+
+    def _get_prompt_modules(self) -> Dict[str, Any]:
+        """Get prompt modules."""
+        return {}
+
+    def _update_prompts(self, **prompts: BasePromptTemplate) -> None:
+        """Update prompts."""
```

Comment:
Should we really provide default values here? Seems nice to force subclasses to properly implement these?

Reply:
yeah we could just do for all query engines + response synthesizers at once